src/mlpack/core/cv/meta_info_extractor.hpp

```diff
@@ -38,8 +38,14 @@ template<typename MatType,
          bool NumClasses>
 struct TrainForm;
 
-template<typename PT, typename WT, typename... SignatureParams>
-struct TrainFormBase
+// Due to an internal MSVC compiler bug we can't use two parameter packs.
+// So we have to write multiple TrainFormBase forms.
```

Comment:
If Microsoft issues a patch update, we might consider putting this in an `#ifdef` for the Visual Studio compiler version.  The only reason I say that is because if Microsoft is using mlpack as a test for each release of Visual Studio, then we want to make sure that the things that expose bugs get tested so that future versions of Visual Studio won't break on valid C++ code. :+1:

I'm ok with the code either way though, this is just a suggestion.